DopplerCurrencyJob/DopplerCurrencyJob.cs

```diff
@@ -25,27 +29,85 @@ public class DopplerCurrencyJob
+
             var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
 
             if (!currencyDto.Any())
             {
-                _logger.LogInformation("Non-existent currencies for this date, please check for errors.");
-                return new List<CurrencyResponse>();
+                LogAndAppendToStringBuilder(resultBuffer, "Non-existent currencies for this date, please check for errors.");
+                return new DopplerCurrencyJobResponse()
+                {
+                    CurrencyList = new List<CurrencyResponse>(),
+                    Message = resultBuffer.ToString()
+                };
+            }
+
+            LogAndAppendToStringBuilder(resultBuffer, "Currency retrieved by code:");
+            AppendSerializedToStringBuilder(resultBuffer, currencyDto);
+
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Sending currency data to Doppler SAP system.");
+                var result = await _dopplerSapService.SendCurrency(currencyDto);
+
+                if (!result.IsSuccessStatusCode)
+                {
+                    LogAndAppendToStringBuilder(resultBuffer, $"{result.ReasonPhrase}. Error sending currency to Doppler SAP Api.");
+                    taskFailed = true;
+                }
+
+                LogAndAppendToStringBuilder(resultBuffer, "Sent currency data to Doppler SAP system.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when sending currency to Doppler SAP Api. Exception: {e.Message}");
+                taskFailed = true;
             }
 
-            _logger.LogInformation("Sending currency data to Doppler SAP system.");
-            await _dopplerSapService.SendCurrency(currencyDto);
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Insert currency data into Doppler Database.");
+
+                await _dopplerCurrencyService.InsertCurrencyIntoDataBase(currencyDto);
+
+                LogAndAppendToStringBuilder(resultBuffer, "Inserted currency data into Doppler Database.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when Insert currency data into Doppler Database. Exception: {e.Message}");
+                taskFailed = true;
+            }
+
+            if (taskFailed)
+            {
+                throw new DopplerCurrencyJobException(resultBuffer.ToString());
+            }
 
-            _logger.LogInformation("Insert currency data into Doppler Database.");
-            await _dopplerCurrencyService.InsertCurrencyIntoDataBase(currencyDto);
+            return new DopplerCurrencyJobResponse()
+            {
+                CurrencyList = currencyDto,
+                Message = resultBuffer.ToString()
```

Comment:
I'm not sure it shows all the log.information in the public url, this information should be in the logging of the server

Reply:
I understand, but it seems to me that sensitive information was not being displayed, but it would be useful to have information in case of problems.
As mentioned in a daily, public access to the service is already being given and @andresmoschini commented that it could be protected with a login in a future,

Reply:
No entendí

Reply:
En una daily comentaste que estaría bueno proteger el acceso al home de hangfire de jobs con un login, para que no quede accesible como esta ahora. 
Si eso se hace, lo de mostrar información de excepciones o de logueo no importaría.

Reply:
Ahh, ok, pero ahora no lo tenemos.

Se escaparía alguna información sensible si publicamos esto?

Reply:
A mi entender no

Reply:
@nreal podrás confirmar?

Reply:
![image](https://user-images.githubusercontent.com/6796523/100142624-f23a7980-2e72-11eb-8b95-6c23f02647b0.png)
al parecer no se publica datos de conexion de base de datos o passwords

Reply:
@nreal estoy haciendo algunas pruebas, nose porque hace eso en la presentación hangfire.

Reply:
estoy usando la propiedad message de la exception para mandarle el contenido del log y lo que me parece que esta haciendo hangfire al mostrarlo es: Encabezado=Message | Cuerpo=Message+StackTrace, por eso parece duplicado
@andresmoschini alguna idea?

Reply:
>  @andresmoschini alguna idea?

nah, dejalo así, no le hace mal a nadie